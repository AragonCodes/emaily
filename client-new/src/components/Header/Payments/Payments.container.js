import { connect } from 'react-redux';
import PaymentsComponent from './Payments';

import { authOperations } from '../../../domain/auth';

const mapDispatchToProps = (dispatch) => {
	const handleToken = (token) => {
		dispatch(authOperations.handleToken(token))
	};

	return { handleToken };
};

const PaymentsContainer = connect(
	null,
	mapDispatchToProps
)(PaymentsComponent);

export default PaymentsContainer;