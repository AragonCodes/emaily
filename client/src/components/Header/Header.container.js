import { connect } from 'react-redux';
import HeaderComponent from './Header';

function mapStateToProps({ auth }) {
	return { auth };
}

const HeaderContainer = connect(
	mapStateToProps
)(HeaderComponent);

export default HeaderContainer;