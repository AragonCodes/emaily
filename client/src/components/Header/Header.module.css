
@media only screen and (max-width: 600px) {
  .logoText {
		display: none;
	}
}