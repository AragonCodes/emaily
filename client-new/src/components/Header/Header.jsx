import React from 'react';
import { Link } from 'react-router-dom';
import Payments from './Payments';

function Header({ auth }) {
	function renderContent() {
		switch (auth) {
			case null:
				return;
			case false:
				return <li><a href="/auth/google">Login With Google</a></li>;
			default:
				return [
					<li key="1"><Payments /></li>,
					<li key="3" style={{ margin: '0 10px' }}>
						Credits: {auth.credits}
					</li>,
					<li key="2"><a href="/api/logout">Logout</a></li>
				]
		}
	};

	return (
		<nav>
			<div className="nav-wrapper">
				<Link
					to={auth ? '/surveys' : '/'}
					className="left brand-logo"
				>
					<i className="material-icons right">mail_outline</i> Emaily
				</Link>
				<ul className="right">
					{renderContent()}
				</ul>
			</div>
		</nav>
	);
}

export default Header;
