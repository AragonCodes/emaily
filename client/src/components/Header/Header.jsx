import React from 'react';
import { Link } from 'react-router-dom';
import Payments from './Payments';
import MailLogo from '../../assets/mail_white.svg'
import styles from './Header.module.css';

function Header({ auth }) {
	function renderContent() {
		switch (auth) {
			case null:
				return;
			case false:
				return <li><a href="/auth/google">Login With Google</a></li>;
			default:
				return [
					<li key="1"><Payments /></li>,
					<li key="3" style={{ margin: '0 10px' }}>
						Credits: {auth.credits}
					</li>,
					<li key="2"><a href="/api/logout">Logout</a></li>
				]
		}
	};

	return (
		<nav className="red accent-2">
			<div className="nav-wrapper" style={{ padding: '0 1rem' }}>
				<Link
					style={{ height: '100%' }}
					to={auth ? '/surveys' : '/'}
					className="left brand-logo valign-wrapper"
				>
					<span className="valign-wrapper" style={{ height: '100%' }}>
						<span className={styles.logoText}>Emaily</span>
						<img src={MailLogo} style={{ height: '3rem', marginLeft: '.5rem' }} />
					</span>
				</Link>
				<ul className="right nav-mobile">
					{renderContent()}
				</ul>
			</div>
		</nav>
	);
}

export default Header;
