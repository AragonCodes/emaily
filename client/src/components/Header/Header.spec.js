import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import Payments from './Payments';

describe('Header', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Header
				auth={{ credits: 1 }}
			/>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should render Google Login', () => {
		const wrapper = shallow(
			<Header
				auth={false}
			/>
		);
		expect(wrapper.contains(<a href="/auth/google">Login With Google</a>)).toBeTruthy();
	});

	it('should render AddCredits button', () => {
		const wrapper = shallow(
			<Header
				auth={{ credits: 1 }}
			/>
		);
		expect(wrapper.find(Payments).length).toEqual(1);
	});
});