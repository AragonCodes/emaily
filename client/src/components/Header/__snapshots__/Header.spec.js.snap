// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
"<nav className=\\"red accent-2\\">
  <div className=\\"nav-wrapper\\" style={{...}}>
    <Link style={{...}} to=\\"/surveys\\" className=\\"left brand-logo valign-wrapper\\">
      <span className=\\"valign-wrapper\\" style={{...}}>
        <span className=\\"logoText\\">
          Emaily
        </span>
        <img src=\\"mail_white.svg\\" style={{...}} />
      </span>
    </Link>
    <ul className=\\"right nav-mobile\\">
      <li>
        <Connect(Payments) />
      </li>
      <li style={{...}}>
        Credits: 
        1
      </li>
      <li>
        <a href=\\"/api/logout\\">
          Logout
        </a>
      </li>
    </ul>
  </div>
</nav>"
`;
